src/read.c

```diff
@@ -167,18 +167,21 @@ putname(char **buf, size_t buflen, const char *host)
 	p = *buf;
 
 	word = strtok(h, ".");
+	size_t len_word = strlen(word);
```

Comment:
This mixes declarations and code which we try to avoid. Can you move it into the loop, then also only one call is needed?

Reply:
@yarrick, I forked at myself and did a `make test` and I couldn't do any micro-optimization by putting strlen variable out of loop, assert test errors everywhere.